tests/lsub_test.go

```diff
@@ -6,38 +6,93 @@ import (
 
-		c.C(`A003 UNSUBSCRIBE "#news.comp.mail.mime"`)
-		c.OK("A003")
+		c.C(`S001 LSUB "" "*"`)
+		c.S(
+			`* LSUB (\Unmarked) "." "INBOX"`,
+			`* LSUB (\Unmarked) "." "foo.bar"`,
+		)
+		c.S("S001 OK LSUB")
 
-		c.C("B003 CREATE #news.comp.mail.misc")
-		c.OK("B003")
+		c.C(`P001 LSUB "" "%"`)
+		c.S(
+			`* LSUB (\Unmarked) "." "INBOX"`,
+			`* LSUB (\Noselect) "." "foo"`,
+		)
+		c.S("P001 OK LSUB")
 
-		c.C(`A003 UNSUBSCRIBE "#news.comp.mail.misc"`)
-		c.OK("A003")
+		c.C("tag CREATE #news.comp.mail.mime").OK("tag")
 
-		c.C(`A002 LSUB "#news." "comp.mail.*"`)
-		c.OK("A002")
+		c.C(`tag UNSUBSCRIBE "#news.comp.mail.mime"`).OK("tag")
 
-		c.C(`A003 SUBSCRIBE "#news.comp.mail.mime"`)
-		c.OK("A003")
+		c.C("tag CREATE #news.comp.mail.misc").OK("tag")
 
-		c.C(`A004 LSUB "#news." "comp.mail.*"`)
+		c.C(`tag UNSUBSCRIBE "#news.comp.mail.misc"`).OK("tag")
+
+		c.C(`S002 LSUB "#news." "comp.mail.*"`)
+		c.S("S002 OK LSUB")
+
+		c.C(`P002 LSUB "#news." "comp.mail.%"`)
+		c.S("P002 OK LSUB")
+
+		c.C(`tag SUBSCRIBE "#news.comp.mail.mime"`).OK("tag")
+
+		c.C(`S003 LSUB "#news." "comp.mail.*"`)
 		c.S(`* LSUB (\Unmarked) "." "#news.comp.mail.mime"`)
-		c.OK("A004")
+		c.S("S003 OK LSUB")
+
+		c.C(`P003 LSUB "#news." "comp.mail.%"`)
+		c.S(`* LSUB (\Unmarked) "." "#news.comp.mail.mime"`)
+		c.S("P003 OK LSUB")
+
+		c.C(`tag SUBSCRIBE "#news.comp.mail.misc"`).OK("tag")
+
+		c.C(`S004 LSUB "#news." "comp.mail.*"`)
+		c.S(
+			`* LSUB (\Unmarked) "." "#news.comp.mail.mime"`,
+			`* LSUB (\Unmarked) "." "#news.comp.mail.misc"`,
+		)
+		c.OK(`S004`)
+
+		c.C(`P004 LSUB "#news." "comp.mail.%"`)
+		c.S(
+			`* LSUB (\Unmarked) "." "#news.comp.mail.mime"`,
+			`* LSUB (\Unmarked) "." "#news.comp.mail.misc"`,
+		)
+		c.OK(`P004`)
+
+		c.C(`S005 LSUB "#news." "comp.*"`)
+		c.S(
+			`* LSUB (\Unmarked) "." "#news.comp.mail"`,
+			`* LSUB (\Unmarked) "." "#news.comp.mail.mime"`,
+			`* LSUB (\Unmarked) "." "#news.comp.mail.misc"`,
+		)
+		c.OK(`S005`)
+
+		c.C(`P005 LSUB "#news." "comp.%"`)
+		c.S(`* LSUB (\Unmarked) "." "#news.comp.mail"`)
+		c.OK(`P005`)
+	})
+}
+
+func TestLsubSubscribedNotExisting(t *testing.T) {
+	runOneToOneTestWithAuth(t, defaultServerOptions(t, withDelimiter(".")), func(c *testConnection, _ *testSession) {
+		c.C("tag CREATE foo").OK(`tag`)
 
-		c.C(`A005 SUBSCRIBE "#news.comp.mail.misc"`)
-		c.OK("A005")
+		c.C(`A001 LSUB "" "foo"`)
+		c.S(`* LSUB (\Unmarked) "." "foo"`)
+		c.OK(`A001`)
 
-		c.C(`A006 LSUB "#news." "comp.mail.*"`)
-		c.S(`* LSUB (\Unmarked) "." "#news.comp.mail.mime"`,
-			`* LSUB (\Unmarked) "." "#news.comp.mail.misc"`)
-		c.OK(`A006`)
+		c.C(`tag DELETE foo`).OK(`tag`)
 
-		c.C(`A007 LSUB "#news." "comp.%"`)
-		c.S(`* LSUB (\Noselect) "." "#news.comp.mail"`)
-		c.OK(`A007`)
+		c.C(`A002 LSUB "" "foo"`)
+		// TODO: The server MUST NOT unilaterally remove an existing mailbox name
+		// from the subscription list even if a mailbox by that name no
+		// longer exists.
+		//
+		// c.S(`* LSUB (\Noselect) "." "foo"`)
```

Comment:
do we have an issue created for this to be resolved?

Reply:
yup added ticket to comment